File: ui/src/api/walletTracking.ts
import { getFromApi } from "@/api/core";
import type { WalletTrackingState } from "@/types/walletTracking";

export const getWalletTracking = async (): Promise<WalletTrackingState> =>
  getFromApi<WalletTrackingState>("/wallet-tracking");

File: data/src/api/wallet_tracking.py
from typing import Annotated

from fastapi import APIRouter, Depends

from api.dependencies import get_wallet_tracking_repository
from db.wallet_tracking import WalletTrackingRepository
from domain.wallet_tracking import WalletTrackingState

router = APIRouter()


@router.get("/wallet-tracking", response_model=WalletTrackingState)
def get_wallet_tracking(
    repo: Annotated[WalletTrackingRepository, Depends(get_wallet_tracking_repository)],
) -> WalletTrackingState:
    state = repo.get()
    if state is not None:
        return state

    return WalletTrackingState.not_run()
